### PR TITLE
Fixed #28606 -- Deprecated CachedStaticFilesStorage.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/28606

### CODE REVIEW

File: tests/staticfiles_tests/test_storage.py [hunk 1]
```diff
@@ -232,13 +230,15 @@ def test_post_processing_failure(self):
...
 
+@ignore_warnings(category=RemovedInDjango31Warning)
 @override_settings(
     STATICFILES_STORAGE='django.contrib.staticfiles.storage.CachedStaticFilesStorage',
 )
 class TestCollectionCachedStorage(TestHashedFiles, CollectionTestCase):
     """
     Tests for the Cache busting storage
     """
+    @ignore_warnings(category=RemovedInDjango31Warning)
```
Comment (charettes) [inline tests/staticfiles_tests/test_storage.py]: Is this required given the class is already decorated?

File: tests/staticfiles_tests/test_storage.py [hunk 2]
```diff
@@ -282,6 +282,7 @@ def test_cache_key_memcache_validation(self):
         cache_validator.validate_key(cache_key)
         self.assertEqual(cache_key, 'staticfiles:821ea71ef36f95b3922a77f7364670e7')
 
+    @ignore_warnings(category=RemovedInDjango31Warning)
```
Comment (charettes) [inline tests/staticfiles_tests/test_storage.py]: Is this required given the class is already decorated?

File: tests/staticfiles_tests/test_storage.py [hunk 3]
```diff
@@ -299,10 +300,20 @@ def test_corrupt_intermediate_files(self):
...
 
 
-@override_settings(
-    STATICFILES_STORAGE='staticfiles_tests.storage.ExtraPatternsCachedStaticFilesStorage',
-)
-class TestExtraPatternsCachedStorage(CollectionTestCase):
+class TestCachedStaticFilesStorageDeprecation(SimpleTestCase):
+    def test_warning(self):
+        from django.contrib.staticfiles.storage import CachedStaticFilesStorage
+        from django.utils.deprecation import RemovedInDjango31Warning
```
Comment (charettes) [inline tests/staticfiles_tests/test_storage.py]: I guess these imports were moved here for easier removal once the deprecation period ends?

Comment (timgraham) [reply tests/staticfiles_tests/test_storage.py]: Yes

**Pull request merged**